packages/headless/src/controllers/pager/headless-pager.ts

```diff
@@ -57,20 +57,57 @@ const initialStateSchema = new Schema({
   page: new NumberValue({min: 1}),
 });
 
-export type Pager = ReturnType<typeof buildPager>;
+export interface Pager {
```

Comment:
The mass interface edit is certainly gonna painful be but if it's necessary 🤷‍♂️

Reply:
How does it affect intellisense in vscode ?

Is there any major difference, or is it essentially the same ?

Reply:
The intellisense is identical.

Ideally, I would have liked to keep the freedom to use any of typescript's features, but the ts compiler is ahead of the documentation tools atm, which are either unable to resolve the documentation (e.g. api-extractor) or resolve the documentation but output a JSON with all the intermediate steps (typedoc).